src/rgw/rgw_op.cc

```diff
@@ -2214,6 +2215,13 @@ void RGWGetObj::execute(optional_yield y)
   std::unique_ptr<RGWGetObj_Filter> run_lua;
   map<string, bufferlist>::iterator attr_iter;
 
+  bool rgw_perf_counters_cache = s->cct->_conf.get_val<bool>("rgw_perf_counters_cache");
+  std::string labels = ceph::perf_counters::key_create("rgw_get", {{"Bucket", s->bucket_name}, {"User", s->user->get_display_name()}});
+  if(rgw_perf_counters_cache) {
+    perf_counters_cache->add(labels, rgw_get_counters);
+    perf_counters_cache->inc(labels, l_rgw_cache_get_ops, 1);
```

Comment:
i'm not really sure why we need a separate `add()` function instead of building this logic into the other calls like `inc()`/`dec()`/etc. even if `add()` creates a new instance, it's possible for that instance to get evicted before its use in the later calls. by combining the 'add' part with the other calls, they happen under the same lock so the operations are guaranteed to succeed

in `RGWGetObj`, i see that we're making 3 calls on the same instance (for l_rgw_cache_get_ops, l_rgw_cache_get_b, and l_rgw_cache_get_lat). if this is a common case, we might consider adding another cache interface like:
```c++
std::shared_ptr<PerfCounters> get(const std::string& key);
```
this way we'd only need to acquire the mutex for one lookup instead of 3 separate calls. and with `shared_ptr`, it's okay if the cache evicts that entry in the meantime because the instance won't be deleted until the caller is done with it